renderer/reducers/backup.js

```diff
@@ -224,6 +274,54 @@ export const setBackupPathLocal = localPath => {
     localPath,
   }
 }
+/**
+ * setRestoreMode - sets if we are in the restore mode currently.
+ *
+ * @param {boolean} value local filesystem directory URI
+ */
+export const setRestoreMode = value => {
+  return {
+    type: SET_RESTORE_MODE,
+    value: isSCBRestoreEnabled() && value,
+  }
+}
+
+export const queryWalletBackup = (walletId, provider) => async (dispatch, getState) => {
+  const backupDesc = (await dbGet(walletId)) || {}
+  if (backupDesc[provider]) {
+    const state = getState()
+    const nodePub = infoSelectors.nodePubkey(state)
+    const { backupId } = backupDesc[provider]
+    dispatch(send('queryBackup', { walletId, backupId, nodePub, provider }))
+  }
+}
+
+export const queryWalletBackupSuccess = (event, { walletId, backup }) => async dispatch => {
+  try {
+    dispatch(restoreWallet(backup))
+    await setRestoreState(walletId, RESTORE_STATE_COMPLETE)
+  } catch (e) {
+    await setRestoreState(walletId, RESTORE_STATE_ERROR)
+  }
+}
+export const queryWalletBackupFailure = () => async dispatch => {
+  // TODO add intl support
+  dispatch(showError(`Unable to find backup file`))
+}
+
+export const restoreWallet = backup => async dispatch => {
+  try {
+    const grpc = await grpcService
+    const result = await grpc.services.Lightning.restoreChannelBackups({
+      multi_chan_backup: backup,
+    })
+    // TODO add intl support
+    dispatch(showNotification(`Wallet backup imported successfully`))
+    return result
+  } catch (e) {
+    dispatch(showError(`Backup import has failed: incorrect backup format`))
```

Comment:
Can this fail for other reasons? Should we include `e.message` in this to help people debug import failures?

Reply:
added message. We had this convo is slack. eventually, we'd want customized messages here